geode-core/src/test/java/org/apache/geode/internal/cache/LocalRegionUpdateTest.java

```diff
@@ -0,0 +1,151 @@
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.function.Function;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import org.mockito.quality.Strictness;
+
+import org.apache.geode.cache.AttributesFactory;
+import org.apache.geode.cache.DataPolicy;
+import org.apache.geode.cache.InterestPolicy;
+import org.apache.geode.cache.InterestResultPolicy;
+import org.apache.geode.cache.RegionAttributes;
+import org.apache.geode.cache.SubscriptionAttributes;
+import org.apache.geode.cache.client.internal.PoolImpl;
+import org.apache.geode.cache.client.internal.RegisterInterestTracker;
+import org.apache.geode.cache.client.internal.ServerRegionProxy;
+import org.apache.geode.distributed.internal.DSClock;
+import org.apache.geode.distributed.internal.InternalDistributedSystem;
+import org.apache.geode.internal.cache.tier.InterestType;
+
+public class LocalRegionUpdateTest {
+  private InternalDataView internalDataView;
+  private LocalRegion region;
+  private RegisterInterestTracker registerInterestTracker;
+
+  @Rule
+  public MockitoRule mockitoRule = MockitoJUnit.rule().strictness(Strictness.STRICT_STUBS);
+
+  @Before
+  public void setUp() {
+    internalDataView = mock(InternalDataView.class);
+
+    EntryEventFactory entryEventFactory = mock(EntryEventFactory.class);
+    InternalCache cache = mock(InternalCache.class);
+    InternalDistributedSystem internalDistributedSystem = mock(InternalDistributedSystem.class);
+    InternalRegionArguments internalRegionArguments = mock(InternalRegionArguments.class);
+
+    LocalRegion.RegionMapConstructor regionMapConstructor =
+        mock(LocalRegion.RegionMapConstructor.class);
+    Function<LocalRegion, RegionPerfStats> regionPerfStatsFactory = localRegion -> {
+      localRegion.getLocalSize();
+      return mock(RegionPerfStats.class);
+    };
+
+    final PoolImpl poolImpl = mock(PoolImpl.class);
+    SubscriptionAttributes subscriptionAttributes =
+        new SubscriptionAttributes(InterestPolicy.ALL);
+
+
+    when(cache.getInternalDistributedSystem()).thenReturn(internalDistributedSystem);
+    when(internalDistributedSystem.getClock()).thenReturn(mock(DSClock.class));
+
+    when(regionMapConstructor.create(any(), any(), any())).thenReturn(mock(RegionMap.class));
+
+
+    AttributesFactory<Object, Object> regionAttributesFactory = new AttributesFactory<>();
+    regionAttributesFactory.setDataPolicy(DataPolicy.NORMAL);
+    regionAttributesFactory.setPoolName("Pool1");
+    regionAttributesFactory.setConcurrencyChecksEnabled(false);
+    regionAttributesFactory.setSubscriptionAttributes(subscriptionAttributes);
+    RegionAttributes<Object, Object> regionAttributes =
+        regionAttributesFactory.createRegionAttributes();
```

Comment:
Use of deprecated methods here can be avoided by using:
```
    InternalRegionFactory<Object, Object> regionFactory = new InternalRegionFactory<>(cache);
    regionFactory.setDataPolicy(DataPolicy.NORMAL);
    regionFactory.setPoolName("Pool1");
    regionFactory.setConcurrencyChecksEnabled(false);
    regionFactory.setSubscriptionAttributes(subscriptionAttributes);
    RegionAttributes<Object, Object> regionAttributes =
        regionFactory.getCreateAttributes();
```

Reply:
Changing code to not use deprecated classes and methods (even if the replacement classes and methods use the deprecated stuff under the hood) allows all uses of the deprecated code to be removed in one go, by changing the implementation in the replacement classes/methods (if necessary), rather than having to find and fix every place that the deprecated code is used.

I agree that it's not ideal that the deprecated class is used inside `RegionFactory`, but that's not a reason to continue using a class explicitly marked as deprecated when an alternative exists and works just fine here. Unless there is a reason that we *must* use the deprecated class, it should not be used.

Reply:
Done.

Reply:
Thanks for the code.